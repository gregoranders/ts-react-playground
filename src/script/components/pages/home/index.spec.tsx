import React, { RecoilRoot, create, mount } from '@app/testUtils';

import * as TestSubject from '@pages/home';

describe(`${TestSubject.HomePage.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(
      <RecoilRoot>
        <TestComponent />
      </RecoilRoot>,
    );
    expect(testSubject.text()).toBe('Generate Users');
  });
});
