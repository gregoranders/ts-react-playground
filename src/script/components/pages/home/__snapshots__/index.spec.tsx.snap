// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot 1`] = `
Array [
  <button
    disabled={false}
    onClick={[Function]}
  >
    Generate Users [
    0
    ]
  </button>,
  <table>
    <tr>
      <td />
      <td />
      <td />
    </tr>
  </table>,
]
`;
